### DISCUSSION TITLE
Mastech6514 Temperature Reader

### DISCUSSION THREAD

**Discussion by Toro322d** [artisan-roaster-scope/artisan#1197] [Q&A] [answered]
Downloaded latest version of Artisan yesterday. Using on Surface and downloaded driver from Silicon Labs. Connected via USB and established connection according to Mastech. However I am not seeing any BT on Artisan. Followed very specific setup instructions from YouTube video from The Captains Coffee site.
Read somewhere since that I should try an older version of Artisan. Any suggestions?

Thank you

**Comment by poundy** [comment]
Yes, don't install old versions ! They aren't supported, we can't help you with anything about them, and we certainly expect things to work in the latest version. 

I don't have a mastech. I do think this will be driver or port related.
Can you tell us, were you using this device on your Surface before, or are you completely new to Artisan? 
According to the info [here](https://artisan-scope.org/devices/meters/) the Mastech 6514 has an internal USB to serial converter. You need to make sure you have the device set up in Artisan correctly, using the correct COM port and the correct serial parameters. In Windows, you can check what the com port that the meter is using by opening Device Manager. Make sure you have Artisan's device to that. 
Screenshots of Device Manager and Artisan port settings would help, as would your Artisan settings file...

**Comment by Toro322d** [comment]
Hello,
Thanks for the reply. This is my first use of Artisan.
Attached are screenshots. 


Joe Buege 
612-875-2595

**Reply by MAKOMO** [reply]
Nothing attached! You need to come over here to Github and drag-and-drop your screenshots to a reply Write text field.

**Reply by Toro322d** [reply]
Sorry. This is new to me.
![41B30D3F-9821-4609-8D64-740C7D55DB57](https://github.com/artisan-roaster-scope/artisan/assets/135984779/466ef522-e9cf-4e31-b91d-86e721dfeaf2)
![E3F1C530-F9E6-4143-8515-CBDB240552AC](https://github.com/artisan-roaster-scope/artisan/assets/135984779/413acb38-dd78-437e-bed9-175bb97ae0e8)
![E8B467BE-ADDB-4F3F-810E-3D9E98B094D2](https://github.com/artisan-roaster-scope/artisan/assets/135984779/ddbaf554-ef33-4b9d-994b-bdaa8dec84e1)
![198F15D3-917F-4392-8819-C4E2B74D8F01](https://github.com/artisan-roaster-scope/artisan/assets/135984779/82e725a6-5813-4ff0-b84f-964d54e01a9e)
![B6115FEE-45DB-4C31-83F6-A64C122C78C9](https://github.com/artisan-roaster-scope/artisan/assets/135984779/a430bfc7-c088-48a6-ac81-a8b2340f57f5)

**Reply by poundy** [reply]
I'll refer you to the link above again. The Mastech is reported as:
Built in USB to serial converter (19200-8E1)
So that's 19200 baud, not 9600 you have in the driver, and EVEN parity, not None you have set in the driver. 
Then, you need to make sure you set COM4 in Artisan... you didn't show us any of the PORT setup there.

**Reply by Toro322d** [reply]
Made required changes to Ports Configuration in Artisan and Port Settings of driver. See Roaster Scope screen shot with windows showing each
![C2367144-E793-4D1A-9A65-CC0585EE58E7](https://github.com/artisan-roaster-scope/artisan/assets/135984779/828f8d33-845e-4b83-ab6f-e536a21bae51)
 as described. Confirmed USB connection. BT Temp still showing 0.0.
Turned Roaster Scope ON. Half of BT 0.0 disappears again. Message above Roasters Scope title says “MS6514temperature0: 35 bytes received but 18 needed. Turned OFF
Confirmed connection of thermocouple and tried both T1 and T2 terminals. No change.
Reset Artisan and Drive to 9600 baud. BT still 0.0 with Roster Scope OFF and ON with thermocouple T1.
Tried thermocouple in T2. BT temp still 0.0 with Roaster Scope OFF. Turned ON and BT display connects with Mastech.
So I have some success and it would seem I am able to track a roast but still very confused as to what might be going on.

**Reply by poundy** [reply]
It would seem likely that the mastech device has changed the serial parameters the device you have requires. Unsure why they would do that, but that's the only reason I can see that would make sense if it worked on 9600 when our previous intelligence said it was 19200. They may also have changed the format of the data it sends (the "35 bytes received but 18 needed"). Unless you can find more detail about the (new) serial comms, there's only so much we can assist with.  
Please add your artisan settings file here to confirm all the other setup things in your device are correct. Save a copy of them, rename them with a .txt extension, and drop them in a reply here.

**Comment by Toro322d** [comment]
Thank you for your assistance. I will send artisan files. I’ll also update if I can find out anything from Mastech.

Joe Buege 
612-875-2595

**Comment by TreyLMartin** [comment]
Was there is any update on this? I'm having the exact same issue with a surface pro (Windows 10), a Perfect Prime (exact Mastech clone). The Perfect Prime software works just fine, but even closed and Artisan open, I get the exact same errors as the screenshots Toro supplied.

**Reply by poundy** [reply]
if it has never worked in Artisan, then the new communication settings/requirement is the issue. As nobody here has the meter in question, besides you and the other person reporting this post, unless someone can get the proper comms settings/protocol out of the vendor, Artisan can't change anything. How is your serial debug skills?

**Comment by TreyLMartin** [comment]
I have a solution provided by someone on Reddit. They said to try T2 instead of T1 and it worked!   
  

  

  
null
  
  
  
  
  
>

**Comment by Toro322d** [comment]
T2 also worked for meJoe Buege 612-875-2595On Nov 5, 2023, at 3:27 PM, TreyLMartin ***@***.***> wrote:﻿
Was there is any update on this? I'm having the exact same issue with a surface pro (Windows 10), a Perfect Prime (exact Mastech clone). The Perfect Prime software works just fine, but even closed and Artisan open, I get the exact same errors as the screenshots Toro supplied.

**Reply by poundy** [reply]
so your earlier tests on T2 were successful but you reported they weren't? Or was your testing flawed in some other way, because you reported "Confirmed connection of thermocouple and tried both T1 and T2 terminals. No change.".

**Comment by Toro322d** [comment]
And as I continued in the same post, “ Tried thermocouple in T2. BT temp still 0.0 with Roaster Scope OFF. Turned ON and BT display connects with Mastech.So I have some success and it would seem I am able to track a roast but still very confused as to what might be going on.”All roasts since then have consistently worked with the thermocouple plugged in to T2 and not T1.Joe Buege 612-875-2595On Nov 6, 2023, at 5:16 PM, poundy ***@***.***> wrote:﻿
so your earlier tests on T2 were successful but you reported they weren't? Or was your testing flawed in some other way, because you reported "Confirmed connection of thermocouple and tried both T1 and T2 terminals. No change.".

**Comment by roasterdave** [comment, marked as answer]
A note here for clarity.  Artisan has always assigned the Mastech 6514 **T1** Port to **ET**.  The **T2** port is assigned to **BT.**

Users of the Fresh Roast typically disable **ET** in Artisan.  Thus, no **BT** temperature will be seen when the thermocouple is plugged into **T1**.  Plugging the thermocouple into **T2** allows the temperature to be displayed as **BT**.

Remember, for temperatures to show in Artisan the Mastech USB link must be on (**PC LINK** button) and either **ON** or **START** is clicked in Artisan.  Temperatures are not displayed while Artisan is **OFF**.

**Comment by TreyLMartin** [comment]
Thank you for the extra information!   
  

  

  
null
  
  
  
  
  
>

**Comment by TreyLMartin** [comment]
Since I am new, how do you disable ET in Artisan?

- Trey


------ Original Message ------
From "treymart" ***@***.***>
To "artisan-roaster-scope/artisan" 
***@***.***>
Date 11/10/2023 7:59:26 PM
Subject Re: [artisan-roaster-scope/artisan] Mastech6514 Temperature 
Reader (Discussion #1197)

**Comment by roasterdave** [comment]
To disable display of **ET** use the menu Config>> Device>> ET/BT tab.  Uncheck **ET** under **Curves** and **LCDs**.

The [Quick Start Guide](https://artisan-scope.org/docs/quick-start-guide/) is an excellent resource to learn about Artisan.

**Comment by TreyLMartin** [comment]
Thanks, Dave I figured it out and it was off. I've set my settings following the captains corner   
  

  

  
null
  
  
  
  
  
>
